lib/lhm.rb

```diff
@@ -82,15 +83,26 @@ def cleanup_current_run(run, table_name, options = {})
     Lhm::Cleanup::Current.new(run, table_name, connection, options).execute
   end
 
-  def setup(connection)
-    @@connection = Lhm::Connection.new(connection: connection)
+  # Setups DB connection
+  #
+  # @param [ActiveRecord::Base] connection ActiveRecord Connection
+  # @param [Hash] connection_options Optional options (defaults to: empty hash)
+  # @option options [Boolean] :reconnect_with_consistent_host
+  #   Active / Deactivate ProxySQL-aware reconnection procedure (default to: false)
+  def setup(connection, connection_options = {})
+    @@connection = Lhm::Connection.new(connection: connection, options: connection_options)
   end
 
-  def connection
+  # Setups DB connection
+  #
+  # @param [Hash] connection_options Optional options (defaults to: empty hash)
+  # @option options [Boolean] :reconnect_with_consistent_host
+  #   Active / Deactivate ProxySQL-aware reconnection procedure (default to: false)
+  def connection(connection_options = {})
     @@connection ||=
       begin
         raise 'Please call Lhm.setup' unless defined?(ActiveRecord)
-        ActiveRecord::Base.connection
+        Lhm::Connection.new(connection: ActiveRecord::Base.connection, options: connection_options)
```

Comment:
Nit: You don't need to prefix with main module name here. Something like `Connection` should work instead of `Lhm::Connection`. Just to maintain with the consistency of the code.